langchain4j-azure-aca-dynamic-sessions/src/main/java/dev/langchain4j/azure/aca/dynamicsessions/SessionsREPLTool.java

```diff
@@ -0,0 +1,566 @@
+    }
+
+    /**
+     * Constructs a new SessionsREPLTool with specified parameters.
+     *
+     * @param poolManagementEndpoint the pool management endpoint URL
+     * @param sessionId the session ID
+     * @param sanitizeInput whether to sanitize the input code
+     */
+    public SessionsREPLTool(String poolManagementEndpoint, String sessionId, boolean sanitizeInput) {
+        this.poolManagementEndpoint = poolManagementEndpoint;
+        this.sessionId = sessionId;
+        this.sanitizeInput = sanitizeInput;
+        this.nativeHttpClient = java.net.http.HttpClient.newBuilder().build();
+
+        // The loadFactories() method returns a Collection, not a List,
+        // so we need to use the iterator instead of get(0)
+        Collection<HttpClientBuilder> builders = ServiceHelper.loadFactories(HttpClientBuilder.class);
+        if (builders.isEmpty()) {
+            throw new IllegalStateException(
+                    "No HttpClientBuilder implementation found. Make sure you have a proper implementation on the classpath.");
+        }
+        this.langchainHttpClient = builders.iterator().next().build(); // Use the langchain4j HTTP client abstraction
+
+        this.credential = new DefaultAzureCredentialBuilder().build();
+    }
+
+    /**
+     * Protected constructor for testing purposes that allows injecting dependencies.
+     *
+     * @param poolManagementEndpoint the pool management endpoint URL
+     * @param sessionId the session ID
+     * @param sanitizeInput whether to sanitize the input code
+     * @param httpClient a pre-configured HTTP client
+     * @param credential a pre-configured credential
+     */
+    protected SessionsREPLTool(
+            String poolManagementEndpoint,
+            String sessionId,
+            boolean sanitizeInput,
+            HttpClient httpClient,
+            DefaultAzureCredential credential) {
+        this.poolManagementEndpoint = poolManagementEndpoint;
+        this.sessionId = sessionId;
+        this.sanitizeInput = sanitizeInput;
+        this.nativeHttpClient = java.net.http.HttpClient.newBuilder().build();
+        this.langchainHttpClient = httpClient;
+        this.credential = credential;
+    }
+
+    // Method is now a no-op as langchain4j HTTP client handles resource cleanup internally
+    // public void shutdown() {
+    // No manual cleanup needed for the HTTP client
+    // }
+
+    /**
+     * Executes the input code and returns the result.
+     *
+     * @param input the code or query to execute
+     * @return the execution result as a JSON string
+     */
+    @Tool(name = "sessions_REPL")
+    public String use(String input) {
+        Map<String, Object> response = executeCode(input);
+
+        Object result = response.get("result");
+        if (result instanceof Map<?, ?>) {
+            Map<?, ?> resultMap = (Map<?, ?>) result;
+            if ("image".equals(resultMap.get("type")) && resultMap.containsKey("base64_data")) {
```

Comment:
```suggestion
            if ("image".equals(resultMap.get("type"))) {
```

As you remove it afterwards, the check is always OK anyway